docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,394 @@
+# Derive minimal ServiceAccount required for ClusterExtension Installation and Management
+
+OLMv1 does not provide cluster admin privileges by default for installing cluster extensions. This document will explain how to derive the required RBAC configuration for a provided service account"
+
+The ServiceAccount is specified in the ClusterExtension manifest as shown below:
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      version: 0.6.0
+  install:
+    namespace: argocd
+    serviceAccount:
+      name: argocd-installer
+```
+
+The initial stable version (v1.0.0) only supports FBC catalogs containing registry+v1 bundles. OLMv1 will not support all OLMv0 content. OLMv1 will only support bundles that meet the following criteria:
+* AllNamespaces install mode is enabled
+* No dependencies on other packages or GVKs
+* No webhooks
+* Does not make use of the OperatorConditions API
+
+### Required RBAC
+
+You can determine the specifics of these permissions by referencing the bundle of the ClusterExtension you want to install. The service account must have the following permissions:
+
+* ClusterRole with all the roles specified in the bundle ClusterServiceVersion. This includes all the
+  rules defined in the CSV under `.spec.install.clusterPermissions` and `.spec.install.permissions`
+* ClusterRole to create and manage CustomResourceDefinitions
+* Update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+* Permissions to create the controller deployment, this corresponds to the rules to manage the
+  deployment defined in the ClusterServiceVersion
+* Permissions to create the other manifest objects, rules to manage any other cluster-scoped resources
+  shipped with the bundle
+* Rules to manage any other namespace-scoped resources 
+* Permissions to create the necessary roles and rolebindings for the controller to be able to perform its job
+* Get, list, watch, update, patch, delete the specific resources that get created/
```

Comment:
Could this be simplified to something like:
---
- `create`, `list`, `watch` verbs for all resources that are a part of the install (cannot be scoped to specific resource names)
    - Permissions for cluster-scoped resources require a `ClusterRole` + `ClusterRoleBinding`
    - Permissions for namespace-scoped resources can be given using `ClusterRole` + `ClusterRoleBinding`, `ClusterRole` + `RoleBinding`, or `Role` + `RoleBinding`
- `get`, `update`, `patch`, `delete` verbs for all resources that are a part of the install (can be scoped to specific resource names)
    - Permissions for cluster-scoped resources require a `ClusterRole` + `ClusterRoleBinding`
    - Permissions for namespace-scoped resources can be given using `ClusterRole` + `ClusterRoleBinding`, 
- Permissions to create any `ClusterRole`, `ClusterRoleBinding`, `Role`, `RoleBinding` resources required by the extension. This means one of:
    - All the same rules defined in the `ClusterRole` and `Role` resources
    - `escalate` and `bind` verbs for `ClusterRole`, `ClusterRoleBinding`, `Role`, `RoleBinding` resources
---

Note: I _think_ this is an exhaustive list of the permissions required, but it is totally possible I may have missed something. This should be interpreted as a general structure suggestion rather than an exact replacement.

Reply:
updated